oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -235,7 +235,19 @@ protected ExternalAccountCredentials(ExternalAccountCredentials.Builder builder)
     this.serviceAccountImpersonationUrl = builder.serviceAccountImpersonationUrl;
     this.clientId = builder.clientId;
     this.clientSecret = builder.clientSecret;
-    this.tokenUrl = builder.tokenUrl == null ? DEFAULT_TOKEN_URL : builder.tokenUrl;
+
+    if (builder.tokenUrl == null) {
+      try {
+        this.tokenUrl = DEFAULT_TOKEN_URL.replace("{UNIVERSE_DOMAIN}", this.getUniverseDomain());
+      } catch (IOException e) {
+        // Throwing an IOException would be a breaking change, so wrap it here.
+        throw new IllegalStateException(
+            "Error occurred when attempting to retrieve universe domain.", e);
+      }
+    } else {
+      this.tokenUrl = builder.tokenUrl;
+    }
```

Comment:
```suggestion
    this.tokenUrl = builder.tokenUrl;
    if (this.tokenUrl == null) {
      try {
        this.tokenUrl = DEFAULT_TOKEN_URL.replace("{UNIVERSE_DOMAIN}", this.getUniverseDomain());
      } catch (IOException e) {
        // Throwing an IOException would be a breaking change, so wrap it here.
        // This should not happen for this credential type.
        throw new IllegalStateException(
            "Error occurred when attempting to retrieve universe domain.", e);
      }
    }
```

Reply:
had to add the else back in since tokenUrl is final so can only be set once.